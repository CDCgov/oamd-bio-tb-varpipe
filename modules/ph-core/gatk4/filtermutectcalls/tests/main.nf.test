nextflow_process {

    name "Test Process GATK4_FILTERMUTECTCALLS"
    script "../main.nf"
    process "GATK4_FILTERMUTECTCALLS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_phcore"
    tag "gatk4"
    tag "gatk4/filtermutectcalls"

    test("human - vcf - base") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.tbi'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.stats'),
                    [],
                    [],
                    [],
                    []
                    ]

                input[1] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta')
                    ]
                input[2] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai')
                    ]
                input[3] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.dict')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions_base") },
                { assert path(process.out.vcf.get(0).get(1)).linesGzip.contains("##fileformat=VCFv4.2") },
                { assert path(process.out.tbi.get(0).get(1)).linesGzip.toString().contains("TBI")},
                { assert snapshot(file(process.out.stats.get(0).get(1)).readLines()[0]).match() }
            )
        }

    }

    test("human - vcf - with-files") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.tbi'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.stats'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/test_test2_paired_mutect2_calls.artifact-prior.tar.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/test_test2_paired.segmentation.table'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/test_test2_paired.contamination.table'),
                    []
                    ]

                input[1] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta')
                    ]
                input[2] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai')
                    ]
                input[3] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.dict')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions_with-files") },
                { assert path(process.out.vcf.get(0).get(1)).linesGzip.contains("##fileformat=VCFv4.2") },
                { assert path(process.out.tbi.get(0).get(1)).linesGzip.toString().contains("TBI")},
                { assert snapshot(file(process.out.stats.get(0).get(1)).readLines()[0..5]).match() }
            )
        }

    }

    test("human - vcf - use-val") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.tbi'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.stats'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/test_test2_paired_mutect2_calls.artifact-prior.tar.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/test_test2_paired.segmentation.table'),
                    [],
                    '20.0'
                    ]

                input[1] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta')
                    ]
                input[2] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai')
                    ]
                input[3] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.dict')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions_use-val") },
                { assert path(process.out.vcf.get(0).get(1)).linesGzip.contains("##fileformat=VCFv4.2") },
                { assert path(process.out.tbi.get(0).get(1)).linesGzip.toString().contains("TBI")},
                { assert snapshot(file(process.out.stats.get(0).get(1)).readLines()[0]).match() }
            )
        }

    }

    test("human - vcf - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.tbi'),
                    file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.stats'),
                    [],
                    [],
                    [],
                    []
                    ]

                input[1] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta')
                    ]
                input[2] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai')
                    ]
                input[3] = [ [ id:'genome' ], // meta map
                        file(params.nfcore_modules_testdata_base_path + '/genomics/homo_sapiens/genome/chr21/sequence/genome.dict')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
