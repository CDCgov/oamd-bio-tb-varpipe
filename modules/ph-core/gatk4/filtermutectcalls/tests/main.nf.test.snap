{
    "human - vcf - use-val": {
        "content": [
            "#<METADATA>Ln prior of deletion of length 10=-20.72326583694641"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-15T12:46:03.887912"
    },
    "versions_with-files": {
        "content": [
            [
                "versions.yml:md5,b61137c9ea27fbd094fb32a621ef6dd2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:15:11.851830078"
    },
    "human - vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz.filteringStats.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,b61137c9ea27fbd094fb32a621ef6dd2"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz.filteringStats.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b61137c9ea27fbd094fb32a621ef6dd2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:15:32.003608744"
    },
    "versions_use-val": {
        "content": [
            [
                "versions.yml:md5,b61137c9ea27fbd094fb32a621ef6dd2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:15:22.759432214"
    },
    "versions_base": {
        "content": [
            [
                "versions.yml:md5,b61137c9ea27fbd094fb32a621ef6dd2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:14:59.925089946"
    },
    "human - vcf - with-files": {
        "content": [
            [
                "#<METADATA>Ln prior of deletion of length 10=-20.72326583694641",
                "#<METADATA>Ln prior of deletion of length 9=-20.72326583694641",
                "#<METADATA>Ln prior of deletion of length 8=-20.72326583694641",
                "#<METADATA>Ln prior of deletion of length 7=-20.72326583694641",
                "#<METADATA>Ln prior of deletion of length 6=-20.72326583694641",
                "#<METADATA>Ln prior of deletion of length 5=-20.72326583694641"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-15T12:45:43.699286"
    },
    "human - vcf - base": {
        "content": [
            "#<METADATA>Ln prior of deletion of length 10=-20.72326583694641"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-15T12:47:39.949405"
    }
}
